src/nvim/api/buffer.c

```diff
@@ -757,6 +756,106 @@ void nvim_buf_set_text(uint64_t channel_id, Buffer buffer, Integer start_row, In
   try_end(err);
 }
 
+/// Gets a range from the buffer.
+///
+/// This differs from |nvim_buf_get_lines()| in that it allows retrieving only
+/// portions of a line.
+///
+/// Indexing is zero-based. Column indices are end-exclusive.
+///
+/// Prefer |nvim_buf_get_lines()| when retrieving entire lines.
+///
+/// @param channel_id
+/// @param buffer           Buffer handle, or 0 for current buffer
+/// @param start_row        First line index
+/// @param start_column     First column
+/// @param end_row          Last line index
+/// @param end_column       Last column
+/// @param[out] err         Error details, if any
+ArrayOf(String) nvim_buf_get_text(uint64_t channel_id, Buffer buffer,
```

Comment:
I think this could use documentation for the return value. At first sight a reader could be confused thinking this returns a string like `line1\nline2\nline3`. (My first thought).